staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/fifo_list.go

```diff
@@ -60,6 +65,9 @@ type fifo interface {
 // goroutines without additional locking or coordination.
 type requestFIFO struct {
 	*list.List
+
+	lock  sync.Mutex
```

Comment:
i don't think we need the lock, right? 
Where the FIFO is used, the calling function (`timeoutOldRequestsAndRejectOrEnqueueLocked` in this PR) already has acquired a lock on the queueset, so no other goroutine is accessing the queues at that point.

Reply:
yeah - good point